tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -88,7 +93,7 @@ public DeferredTracer(Optional<String> operation) {
         this(operation.orElse(DEFAULT_OPERATION));
     }
 
-    public DeferredTracer(String operation) {
+    public DeferredTracer(String operation, Map<String, String> metadata) {
```

Comment:
I opted against including a TagRecorder constructor for DeferredTracer because there's no guarantee that implementations will be serializable, which is important for spark jobs. Additionally, these are most frequently used for executors where the tags are constant across a single executor instance..

Reply:
The unit test validates that objects can be serialized, but it wouldn't fail if we happened to pass serializable implementations. We wouldn't want to serialize the entire raw input in that case either... serialization is way too subtle, and I've enjoyed not having to work with it for the last year ;-)

An error-prone rule for serializable data would be nifty, I've been burned by many serialization failures! Not sure how we'd set up such a check to avoid false positives though.